test/framework/options.py

```diff
@@ -1065,18 +1069,136 @@ def check_copied_files():
+        bat_ec = os.path.join(self.test_prefix, 'bat.eb')
+        args[-1] = bat_ec
+        stdout = self.mocked_main(args)
+
+        regex = re.compile(r"%s copied to .*/bat.eb" % ec_filename)
+        self.assertTrue(regex.search(stdout), "Pattern '%s' should be found in: %s" % (regex.pattern, stdout))
+
+        self.assertTrue(os.path.exists(bat_ec))
+        self.assertTrue("name = 'bat'" in read_file(bat_ec))
+
+        change_dir(cwd)
+        remove_dir(test_working_dir)
+        mkdir(test_working_dir)
+        change_dir(test_working_dir)
+
+        # test copying of a patch file from a PR via --copy-ec to current directory
+        patch_fn = 'bat-0.3.3-fix-pyspark.patch'
+        args = ['--copy-ec', '--from-pr', '8007', patch_fn, '.']
+        stdout = self.mocked_main(args)
+
+        self.assertEqual(os.listdir(test_working_dir), [patch_fn])
+        patch_path = os.path.join(test_working_dir, patch_fn)
+        self.assertTrue(os.path.exists(patch_path))
+        self.assertTrue(is_patch_file(patch_path))
+        remove_file(patch_path)
+
+        # test the same thing but where we don't provide a target location
+        change_dir(test_working_dir)
+        args = ['--copy-ec', '--from-pr', '8007', ec_filename]
+        stdout = self.mocked_main(args)
+
+        regex = re.compile(r"%s copied to .*/%s" % (ec_filename, os.path.basename(test_working_dir)))
+        self.assertTrue(regex.search(stdout), "Pattern '%s' should be found in: %s" % (regex.pattern, stdout))
+
+        self.assertEqual(os.listdir(test_working_dir), [ec_filename])
+        self.assertTrue("name = 'bat'" in read_file(os.path.join(test_working_dir, ec_filename)))
+
+        # also test copying of patch file to current directory (without specifying target location)
+        change_dir(test_working_dir)
+        args = ['--copy-ec', '--from-pr', '8007', patch_fn]
+        stdout = self.mocked_main(args)
+
+        regex = re.compile(r"%s copied to .*/%s" % (patch_fn, os.path.basename(test_working_dir)))
+        self.assertTrue(regex.search(stdout), "Pattern '%s' should be found in: %s" % (regex.pattern, stdout))
+
+        self.assertEqual(sorted(os.listdir(test_working_dir)), sorted([ec_filename, patch_fn]))
+        self.assertTrue(is_patch_file(os.path.join(test_working_dir, patch_fn)))
+
+        change_dir(cwd)
+        remove_dir(test_working_dir)
+
+        # test with only one ec in the PR (final argument is taken as a filename)
+        test_ec = os.path.join(self.test_prefix, 'test.eb')
+        args = ['--copy-ec', '--from-pr', '11521', test_ec]
+        ec_pr11521 = "ExifTool-12.00-GCCcore-9.3.0.eb"
+        stdout = self.mocked_main(args)
+        regex = re.compile(r'.*/%s copied to %s' % (ec_pr11521, test_ec))
+        self.assertTrue(regex.search(stdout), "Pattern '%s' found in: %s" % (regex.pattern, stdout))
+        self.assertTrue(os.path.exists(test_ec))
```

Comment:
Check the contents as well:
```
self.assertTrue("name = 'ExifTool'" in read_file(test_ec))
```

Reply:
@boegel I injected it so we can hit merge on this

Reply:
:+1: